### ISSUE TITLE
syntactic_sugar flags var declarations that can't be fixed

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#928]
I have this line:

```swift
fileprivate var currentIndex: Array<OnboardingPage>.Index?
```

because I want a variable of the same type as the index of an array of `OnboardingPage`s. However, if I change it to `[OnboardingPage].Index?`, I get this from Xcode:

```
path/to/OnboardingViewController.swift:35:51: error: consecutive declarations on a line must be separated by ';'
    fileprivate var currentIndex: [OnboardingPage].Index?
                                                  ^
                                                  ;
path/to/OnboardingViewController.swift:35:51: error: expected declaration
    fileprivate var currentIndex: [OnboardingPage].Index?
                                                  ^
path/to/OnboardingViewController.swift:13:13: note: in declaration of 'OnboardingViewController'
final class OnboardingViewController: UIViewController {
```

I suppose this should also be considered a Swift bug, but SwiftLint shouldn't report issues that can't be fixed. For now, I'll use `// swiftlint:disable:next syntactic_sugar`

**Comment by marcelofabri** [comment]
I agree we should handle this, but I'm curious on why not just using `Int` 😅.

**Comment by ZevEisenberg** [comment]
I was going to say that `Int` wasn't right, but I think that may have been a carryover from Swift 2. I just changed it to `Int?`, and looks like it works, so yay! Thanks for the suggestion. Pretty sure I had a valid reason for this in Swift 2, but who even knows any more.

**Comment by jpsim** [comment]
I think there's a way to further refine the rule to avoid false positives based on the `ASTRule` results, but that would require more research than I have time to do at the moment.
